grpc/src/test/java/com/linecorp/armeria/it/grpc/GrpcHttpJsonTranscodingServiceAnnotatedAuthServiceTest.java

```diff
@@ -0,0 +1,160 @@
+
+    private final HttpJsonTranscodingTestServiceBlockingStub gRpcClient =
+            GrpcClients.newClient(gRpcUri, HttpJsonTranscodingTestServiceBlockingStub.class);
+
+    @Test
+    void testAuthenticatedRpcMethod() throws Exception {
+        final Transcoding.GetMessageRequestV1 requestMessage = Transcoding.GetMessageRequestV1.newBuilder()
+                .setName("messages/1").build();
+        final Throwable exception = assertThrows(Throwable.class,
+                () -> gRpcClient.getMessageV1(requestMessage).getText());
+        assertThat(exception).isInstanceOf(StatusRuntimeException.class);
+        assertThat(((StatusRuntimeException) exception).getStatus().getCode())
+                .isEqualTo(Status.UNAUTHENTICATED.getCode());
+
+        final Metadata metadata = new Metadata();
+        metadata.put(Metadata.Key.of(TEST_CREDENTIAL_KEY, Metadata.ASCII_STRING_MARSHALLER),
+                "some-credential-string");
+        final Transcoding.Message result = gRpcClient.withInterceptors(
+                MetadataUtils.newAttachHeadersInterceptor(metadata)
+        ).getMessageV1(requestMessage);
+        assertThat(result.getText()).isEqualTo("messages/1");
+    }
+
+    @Test
+    void testAuthenticatedHttpJsonTranscoding() throws Exception {
+        final AggregatedHttpResponse failResponse = webClient.get("/v1/messages/1");
+        assertThat(failResponse.status()).isEqualTo(HttpStatus.UNAUTHORIZED);
+
+        final AggregatedHttpResponse successResponse = webClient
+                .execute(RequestHeaders.of(HttpMethod.GET, "/v1/messages/1",
+                        TEST_CREDENTIAL_KEY, "some-credential-string"));
+        final JsonNode root = mapper.readTree(successResponse.contentUtf8());
+        assertThat(root.get("text").asText()).isEqualTo("messages/1");
+    }
+
+    private static class AuthenticatedHttpJsonTranscodingTestService
+            extends HttpJsonTranscodingTestServiceImplBase {
+        @Override
+        @Authenticate
+        public void getMessageV1(Transcoding.GetMessageRequestV1 request,
+                                 StreamObserver<Transcoding.Message> responseObserver) {
+            responseObserver.onNext(Transcoding.Message.newBuilder().setText(request.getName()).build());
+            responseObserver.onCompleted();
+        }
+    }
+
+    @DecoratorFactory(AuthServiceDecoratorFactoryFunction.class)
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target({ ElementType.TYPE, ElementType.METHOD })
+    private @interface Authenticate{}
+
+    private static class AuthServiceDecoratorFactoryFunction implements DecoratorFactoryFunction<Authenticate> {
+        @NotNull
+        @Override
+        public Function<? super HttpService, ? extends HttpService>
+        newDecorator(@NotNull Authenticate parameter) {
+            return AuthService.newDecorator(
+                    new TestAuthorizer()
+            );
+        }
+    }
```

Comment:
- We don't use `@NotNull` explicitly. Instead, we use `@NonNullByDefault` at the package level.
- How about making it more readable?
```suggestion
    private static class AuthServiceDecoratorFactoryFunction implements DecoratorFactoryFunction<Authenticate> {
        @Override
        public Function<? super HttpService, ? extends HttpService> newDecorator(Authenticate parameter) {
            return AuthService.newDecorator(new TestAuthorizer());
        }
    }
```